types/types.cc

```diff
@@ -350,7 +350,14 @@ static uint32_t serialize(sstring_view input, int64_t days) {
 }
 uint32_t simple_date_type_impl::from_sstring(sstring_view s) {
     char* end;
+    errno = 0;
     auto v = std::strtoll(s.begin(), &end, 10);
+     if (errno == ERANGE) {
+        throw marshal_exception(format("Unable to make unsigned int (for date): the value entered is too large (more than 64 bits)"));
```

Comment:
You called format(), but didn't use any format string. I assume you intended to print the number that caused the error in the message, something like Cassandra:
```
throw marshal_exception(format("Unable to ...: '{}'", s))
```